zookeeper-api/src/main/java/org/apache/helix/zookeeper/zkclient/ZkClient.java

```diff
@@ -1172,13 +1176,32 @@ private void fireNewSessionEvents() {
     }
     final String sessionId = getHexSessionId();
     for (final IZkStateListener stateListener : _stateListener) {
-      _eventThread.send(new ZkEventThread.ZkEvent("New session event sent to " + stateListener, sessionId) {
+      _eventThread
+          .send(new ZkEventThread.ZkEvent("New session event sent to " + stateListener, sessionId) {
 
-        @Override
-        public void run() throws Exception {
-          stateListener.handleNewSession(sessionId);
-        }
-      });
+            @Override
+            public void run() throws Exception {
+              if (_syncOnNewSession) {
+                //System.out.println("syncOnNewSession with sessionID:" + sessionId);
+                LOG.info("syncOnNewSession with sessionId {}", sessionId);
+                final ZkConnection zkConnection = (ZkConnection) getConnection();
+                if (zkConnection == null || zkConnection.getZookeeper() == null) {
```

Comment:
I think zkConnection is not possible to be null here. Maybe zkConnection.getZookeeper() won't be null as well.
But check it is a valid thing to do.
One point is that line 1193 should use the same ZK object as it is checked here. Or it might become null later.

Reply:
I don't think this null check is necessary, as zkConnection/zk won't be null once zkClient object is constructed. Only chance zk being null is this zkClient.close() is called, then zk is null. I would say this may be overcheck for nulls.

Reply:
removed.